scripts/release/publish.js

```diff
@@ -24,25 +24,38 @@ const ARGS = [
 
 const { stderr, stdin, stdout } = process;
 
-const RELEASE_BRANCH_RE = /^(master|((winter|spring|summer)\d+))$/;
-
 try {
     if (!isCI) {
         console.error('This script is only meant to run in CI.');
         process.exit(1);
     }
 
-    // `git branch --show-current` is only available as of git@2.22.0
-    const currentBranch = execa.sync('git', ['rev-parse', '--abbrev-ref', 'HEAD'], { shell: true })
-        .stdout;
-    if (!RELEASE_BRANCH_RE.test(currentBranch)) {
-        console.error(
-            `Invalid branch: "${currentBranch}". This script is only meant to run in a release branch.`
-        );
+    // We can't use a straightforward git command like `git branch --show-current` because we've
+    // checked out a git tag and we're in a detached HEAD state. The following logic attempts to
+    // verify that this tag points to a commit contained by a release branch.
+
+    // All branches that contain the current commit. Since we only deal with release commits, the
+    // only scenario where this might be an issue is if we try to publish a release commit from
+    // master after CLCO because the commit will show up in two branches.
+    const branches = execa.commandSync('git branch --all --contains').stdout;
```

Comment:
I think the easiest way to access the current branch is via `process.env.CIRCLE_BRANCH`: https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables

Reply:
Circleci runs this script runs on tags so we're in a detached HEAD state and there is no current branch.

Reply:
Good call.